server/mergin/sync/workspace.py

```diff
@@ -281,3 +281,55 @@ def update_project_members(
     def access_requests_query():
         """Project access base query"""
         return AccessRequest.query.join(Project)
+
+    @staticmethod
+    def project_access(project: Project) -> List[Dict]:
+        """
+        Project access users overview
+        """
+        global_role = None
+        accesses = (
```

Comment:
I'd simplify a logic bit here: e.g. global_users: all or none and direct users from project.access. For global users set role from GLOBAL flag, for direct users we should have a helper already to get project role (combined project and workspace role).